designer/server/src/common/helpers/auth/user-session.js

```diff
@@ -25,25 +27,33 @@ export function createUser(credentials) {
 }
 
 /**
- * @param {Request} request
+ * @param {Request<{ AuthArtifactsExtra: AuthArtifacts }>} request
  */
 export async function createUserSession(request) {
   const { auth, server } = request
+  const { artifacts, credentials } = auth
 
-  // Optionally create user object (e.g. signed in token but no session)
-  const user = !getUser(auth.credentials)
-    ? createUser(auth.credentials)
-    : auth.credentials.user
+  // Patch missing properties using Bell artifacts
+  credentials.idToken = artifacts.id_token
```

Comment:
I've added types for `AuthArtifacts` so we can pull `artifacts.id_token` from Bell